lib/purgatory/purgatory.rb

```diff
@@ -47,6 +47,7 @@ def approve!(approver = nil)
     return false if approved? 
     if soul_with_changes.save
       soul.send(performable_method[:method],*performable_method[:args]) if performable_method
+      return false if soul.errors.any?
```

Comment:
I guess I'd have to see where/why you're using this in the Financeit code. But perhaps you can just do a check there if `purgatory.soul.errors.any?`?